plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, List, MutableMapping, Optional
+
+import optuna
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    RangeSweep,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    BaseDistribution,
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+from optuna.samplers import CmaEsSampler, RandomSampler, TPESampler
+
+from .config import DistributionConfig, DistributionType, OptunaConfig
+
+log = logging.getLogger(__name__)
+
+
+def create_optuna_distribution_from_config(
+    config: MutableMapping[str, Any]
+) -> BaseDistribution:
+    kwargs = dict(config)
+    if isinstance(config["type"], str):
+        kwargs["type"] = DistributionType[config["type"]]
+    param = DistributionConfig(**kwargs)
+    if param.type == DistributionType.categorical:
```

Comment:
I might be missing the full context but if `param.type` is a `type` probably `isinstance` will be better than `==`

Reply:
The name may be confusing, but `param.type` is not a `type` but an enum value. Do you think I should rename the property name?

Reply:
No its okay :)